.github/workflows/test-sui.yaml

```diff
@@ -139,7 +139,55 @@ jobs:
 
       ###### Command: Multisig ######
 
-      # TODO: Add multisig tests
+      - name: Init Multisig
+        run: |
+          # Create new addresses
+          sui client new-address secp256k1 multisig1
+          sui client new-address secp256k1 multisig2
+
+          # Export keys and addresses
+          KEY_1=$(sui keytool export --key-identity multisig1 --json | jq -r .key.publicBase64Key)
+          KEY_2=$(sui keytool export --key-identity multisig2 --json | jq -r .key.publicBase64Key)
+          ADDRESS_1=$(sui keytool export --key-identity multisig1 --json | jq -r .key.suiAddress)
+          ADDRESS_2=$(sui keytool export --key-identity multisig2 --json | jq -r .key.suiAddress)
+
+          # Get multisig address
+          MULTISIG_ADDRESS=$(sui keytool multi-sig-address --pks $KEY_1 $KEY_2 --weights 1 1 --threshold 1 --json | jq -r .multisigAddress)
+
+          # Initialize multisig
+          node sui/multisig.js --action init --threshold 1 --base64PublicKeys $KEY_1 $KEY_2 --schemeTypes secp256k1 secp256k1
+
+          # Faucet operations
+          node sui/faucet.js --recipient $ADDRESS_1
+          node sui/faucet.js --recipient $ADDRESS_2
+          node sui/faucet.js --recipient $MULTISIG_ADDRESS
+
+          # Set environment variables
+          echo "KEY_1=$KEY_1" >> $GITHUB_ENV
+          echo "KEY_2=$KEY_2" >> $GITHUB_ENV
+          echo "ADDRESS_1=$ADDRESS_1" >> $GITHUB_ENV
+          echo "ADDRESS_2=$ADDRESS_2" >> $GITHUB_ENV
+          echo "MULTISIG_ADDRESS=$MULTISIG_ADDRESS" >> $GITHUB_ENV
+
+      - name: Transfer Upgrade Cap to Multisig Address
+        run: |
+          upgrade_cap=$(cat axelar-chains-config/info/local.json | jq -r '.sui.contracts.AxelarGateway.objects.UpgradeCap')
+          node sui/transfer-object.js --objectId $upgrade_cap --recipient $MULTISIG_ADDRESS
+
+      - name: Generate Unsigned Tx File
+        run: |
+          node sui/deploy-contract.js upgrade AxelarGateway any_upgrade --offline --txFilePath ./tx-upgrade.json --sender $MULTISIG_ADDRESS
+
+      - name: Sign Tx File with Multisig Signer
+        run: |
+          pk_1=$(sui keytool export --key-identity multisig1 --json | jq .exportedPrivateKey | sed 's/"//g')
+          pk_2=$(sui keytool export --key-identity multisig2 --json | jq .exportedPrivateKey | sed 's/"//g')
+          node sui/multisig.js --action sign --txBlockPath ./tx-upgrade.json --signatureFilePath signature-1.json --offline --privateKey $pk_1
+          node sui/multisig.js --action sign --txBlockPath ./tx-upgrade.json --signatureFilePath signature-2.json --offline --privateKey $pk_2
+
+      - name: Submit Signed Tx File
+        run: |
+          node sui/multisig.js --txBlockPath ./tx-upgrade.json --signatureFilePath ./combined.json --action combine --signatures signature-1.json signature-2.json
```

Comment:
Test if gateway approve still works post upgrade

Reply:
Is the contract address being updated to the new package id after upgrade?

Reply:
@milapsheth Yes, it'll be upgraded to the new package id. Approval works with either the upgraded contract or the original contract. In this case, should we implement some mechanism to guard access to all prior versions of the package, because they're still exist on-chain?

Reply:
Foivos is looking into the upgrade mechanism to create a design for our contracts. We want some methods to be accessible on the old contract potentially, and some to not be depending on upgrade versioning